From 428540c4666692b71246adc33397df107d4671c7 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Wed, 31 Oct 2018 16:46:24 +0800
Subject: [PATCH 4966/5242] MLK-19750-4: ARM64: dts: workaround 192kHz
 recording noise issue

commit  0f578434c8134dce7c504666e104a453c51967a1 from
https://source.codeaurora.org/external/imx/linux-imx.git

There is noise issue with 192kHz recoding with ESAI + CS42xx8 on imx8qm
mek board.

This issue is caused by the round trip delay due to longer trace length
on board. After we switch to tx master, rx slave mode, the issue is gone.
so the setting can workaround the issue, the reason is that the bitclock,
frame clock and data is generated from one side, for recording, is from
codec, the asynchronous of clock and data is eliminated.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-imx8qm-mek-m4.dts |    1 +
 arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi  |    2 ++
 2 files changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-m4.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-m4.dts
index 3fd9522..c0eef8a 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-m4.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-m4.dts
@@ -36,6 +36,7 @@
 		VLS-supply = <&reg_audio>;
 		VLC-supply = <&reg_audio>;
 		reset-gpio = <&gpio4 25 1>;
+		fsl,txs-rxm;
 		status = "okay";
 	};
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
index 8181f2f..60aa2f3 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
@@ -646,6 +646,7 @@
 			<&clk IMX8QM_AUD_ESAI_0_EXTAL_IPG>;
 	assigned-clock-parents = <&clk IMX8QM_AUD_ACM_AUD_PLL_CLK0_CLK>;
 	assigned-clock-rates = <0>, <786432000>, <49152000>, <12288000>, <49152000>;
+	fsl,txm-rxs;
 	status = "okay";
 };
 
@@ -885,6 +886,7 @@
 				<&clk IMX8QM_AUD_ACM_AUD_REC_CLK0_DIV>,
 				<&clk IMX8QM_AUD_MCLKOUT0>;
 		assigned-clock-rates = <786432000>, <49152000>, <12288000>, <12288000>;
+		fsl,txs-rxm;
 		status = "okay";
 	};
 };
-- 
1.7.9.5

